# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.h
# Commit: 6fd57b83bb52
# Full Hash: 6fd57b83bb52e9683bade8cbfb5f8ba6575ac6af
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-07 03:55:51
# Regressor Bug: 1582832
# File Overlap Count: 3
# Description:
#   Bug 1582832: Part 1 - Make FrameLoader owner rather than DocShell responsible for discarding a BC. r=nika
#   
#   There are all sorts of lifecycle issues which arise from making DocShell
#   responsible for discarding BrowsingContexts. In this particular bug, we tend
#   to run into them in cases where we create a BrowsingContext for a FrameLoader,
# ==============================================================================

diff -r 28a9e4b075af -r 6fd57b83bb52 docshell/base/nsDocShell.h
--- a/docshell/base/nsDocShell.h	Thu Feb 06 18:35:10 2020 +0000
+++ b/docshell/base/nsDocShell.h	Thu Feb 06 19:07:56 2020 +0000
@@ -477,9 +477,7 @@
   // Clear the document's storage access flag if needed.
   void MaybeClearStorageAccessFlag();
 
-  void SkipBrowsingContextDetach() {
-    mSkipBrowsingContextDetachOnDestroy = true;
-  }
+  void SetWillChangeProcess() { mWillChangeProcess = true; }
 
   // Create a content viewer within this nsDocShell for the given
   // `WindowGlobalChild` actor.
@@ -1355,10 +1353,9 @@
 
   bool mIsFrame : 1;
 
-  // If mSkipBrowsingContextDetachOnDestroy is set to true, then when the
-  // docshell is destroyed, the browsing context will not be detached. This is
-  // for cases where we want to preserve the BC for future use.
-  bool mSkipBrowsingContextDetachOnDestroy : 1;
+  // If mWillChangeProcess is set to true, then when the docshell is destroyed,
+  // we prepare the browsing context to change process.
+  bool mWillChangeProcess : 1;
 
   // Set when activity in this docshell is being watched by the developer tools.
   bool mWatchedByDevtools : 1;